### ISSUE TITLE
AWS::EKS::Cluster - ResourcesVpcConfig-endpointPrivateAccess

### ISSUE THREAD

**Issue by dnascimento** [aws-cloudformation/cloudformation-coverage-roadmap#118]
## 1. Title 
AWS::EKS::Cluster-ResourcesVpcConfig-endpointPrivateAccess

## 2. Scope of request 

AWS::EKS::Cluster-ResourcesVpcConfig supports `SecurityGroupIds` and `SubnetIds` but not `endpointPrivateAccess` and `endpointPublicAccess`. These properties can be created via API but not via CloudFormation.

## 3. Expected behavior

Allow users to set endpointPrivateAccess=True/False and endpointPublicAccess=True/False

```yaml
Type: AWS::EKS::Cluster
Properties: 
  Name: String
  ResourcesVpcConfig: 
    SecurityGroupIds:
        - String
    SubnetIds: 
        - String
    EndpointPrivateAccess: Boolean
    EndpointPublicAccess: Boolean
  RoleArn: String
  Version: String
```

## 4. Suggest specific test cases

Many users do not want to expose their EKS API to public and/or need to expose a private endpoint to EKS. This is a **blocker** for many customers as their security policies don't allow public endpoints. [EKS Cluster Endpoint Access](https://docs.aws.amazon.com/eks/latest/userguide/cluster-endpoint.html)
As alternative, many are using awscli and terraform.

## 5. Helpful Links to speed up research and evaluation

[EKS API Reference](https://docs.aws.amazon.com/eks/latest/APIReference/API_VpcConfigRequest.html#AmazonEKS-Type-VpcConfigRequest-endpointPrivateAccess)
[EKS Cluster Endpoint Access](https://docs.aws.amazon.com/eks/latest/userguide/cluster-endpoint.html)

https://github.com/weaveworks/eksctl/issues/649
https://github.com/weaveworks/eksctl/issues/778
https://github.com/aws/containers-roadmap/issues/242
https://github.com/aws-quickstart/quickstart-amazon-eks/issues/37

## 6. Category
1. Compute (EKS)

**Comment by dnascimento** [comment]
@luiseduardocolon what is the ETA for this feature?

**Comment by rwkarg** [comment]
Would adding CIDR ranges for public endpoints be a separate issue or looped in to this one?

**Comment by luiseduardocolon** [comment]
@dnascimento we don't have an ETA currently.  @rwkarg unsure, I will ask.

**Comment by seb-steuer** [comment]
@luiseduardocolon I would like to add that updating AWS::EKS::Cluster resource to include ResourcesVpcConfig if you only define/change EndpointPrivateAccess, EndpointPublicAccess or PublicAccessCidrs should NOT require replacement. Currently adding ResourcesVpcConfig requires replacement which is not needed to only change values for public/private access in the API/CLI CloudFormation should allow the same level of flexibility.

Kind regards

**Comment by gabegorelick** [comment]
This is an issue from a security standpoint since CloudFormation creates clusters with endpointPublicAccess enabled. Only after creation can you disable that, which means there's a window where your cluster is publicly accessible.

It's also a pain because it can take ~15 minutes (in my testing) to modify `resourcesVpcConfig`, which adds to the total amount of time it takes to provision a cluster (which already takes a long time).

**Comment by xor007** [comment]
+1 we need this

**Comment by lynnnnnnluo** [comment]
Tags logging, end point access will be added in next release

**Comment by aws-tatarkin** [comment]
Hi any updates?

Related: https://github.com/aws/containers-roadmap/issues/242

**Comment by mikestef9** [comment]
This feature is now available through EKS cluster resource

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-eks-cluster.html
